lib/ecto/dev_logger.ex

```diff
@@ -163,6 +164,22 @@ defmodule Ecto.DevLogger do
     |> elem(0)
   end
 
+  defp preprocess_params(metadata) do
+    Enum.zip(metadata.params, metadata.cast_params || [])
```

Comment:
cast_params was added to ecto_sql in v3.9.2
https://github.com/elixir-ecto/ecto_sql/blob/48fc2ad6e8afb022f8454350e23122c3304451d1/CHANGELOG.md?plain=1#L85
so it is better to require ecto_sql `~> 3.10` or to add a condition to return `metadata.params` as is if `cast_params` is absent. Otherwise, on previous ecto versions, the whole result of `preprocess_params` invocation will be always empty list This is just how zip works.

Also, an optimization is possible here: zip+map can be replaced with zip_with/3

Reply:
This is the last comment to address before merging.
Everything else looks good :muscle:

Reply:
I just added an if check, this way ecto does not have be upgraded and the case when `cast_params`  is nil is also resolved.